pkg/codegen/internal/test/testdata/simple-methods-schema-single-value-returns/go/example/foo.go

```diff
@@ -0,0 +1,116 @@
+// *** WARNING: this file was generated by test. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+package example
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type Foo struct {
+	pulumi.ResourceState
+}
+
+// NewFoo registers a new resource with the given unique name, arguments, and options.
+func NewFoo(ctx *pulumi.Context,
+	name string, args *FooArgs, opts ...pulumi.ResourceOption) (*Foo, error) {
+	if args == nil {
+		args = &FooArgs{}
+	}
+
+	var resource Foo
+	err := ctx.RegisterRemoteComponentResource("example::Foo", name, args, &resource, opts...)
+	if err != nil {
+		return nil, err
+	}
+	return &resource, nil
+}
+
+type fooArgs struct {
+}
+
+// The set of arguments for constructing a Foo resource.
+type FooArgs struct {
+}
+
+func (FooArgs) ElementType() reflect.Type {
+	return reflect.TypeOf((*fooArgs)(nil)).Elem()
+}
+
+func (r *Foo) GetKubeconfig(ctx *pulumi.Context, args *FooGetKubeconfigArgs) (pulumi.StringOutput, error) {
+	out, err := ctx.Call("example::Foo/getKubeconfig", args, fooGetKubeconfigResultOutput{}, r)
+	if err != nil {
+		return pulumi.StringOutput{}, err
+	}
+	return out.(fooGetKubeconfigResultOutput).Kubeconfig(), nil
```

Comment:
Would have been possible to omit fooGetKubeconfigResultOutput type entirely? Save space..

Reply:
Not based on the current implementation of the SDK's `Context.Call`, which needs the output:

https://github.com/pulumi/pulumi/blob/230f4ee006995b47751c9ee4f44643ea3427fb82/sdk/go/pulumi/context.go#L331

The SDK itself doesn't know how to lift the single value, so it's handled here in the codegen.

I don't think we have to be too concerned about the size just yet as there are very few uses of methods out there, and even with broader adoption, I doubt it'll be used so much as to cause size issues (famous last words).

Reply:
Ah that's right, these are methods, not functions (of which there's a few).